cli/azd/resources/scaffold/templates/resources.bicept

```diff
@@ -550,6 +549,16 @@ module {{bicepName .Name}} 'br/public:avm/res/app/container-app:0.8.0' = {
             value: storageAccount.outputs.serviceEndpoints.blob
           }
           {{- end}}
+          {{- if $infra.KeyVault}}
```

Comment:
What are your thoughts on if the service should always be allowed access to key vault? Or is this a case of `uses`? I'm debating this myself

Reply:
With these proposed changes, Key Vault is kind of being treated as a special case where `uses: vault` doesn't really do anything differently.

If `uses` determines whether `AZURE_KEY_VAULT_NAME`, `AZURE_KEY_VAULT_ENDPOINT` are set in the container app, then there'd be more cases to handle like what if `vault` isn't explicitly defined in `azure.yaml` a database resource is - would that mean `AZURE_KEY_VAULT_NAME` isn't set?

I could also see making key vault an implicit dependency of container app, then this `{{- if $infra.KeyVault}}` wouldn't be needed.

Reply:
That's fine either way, I think we can leave it as-is for now.